x-pack/plugins/security_solution/public/management/components/artifact_list_page/components/artifact_flyout.tsx

```diff
@@ -219,14 +223,22 @@ export const MaybeArtifactFlyout = memo<ArtifactFlyoutProps>(
       return isEditFlow && !hasItemDataForEdit;
     }, [hasItemDataForEdit, isEditFlow]);
 
+    const clearItem = useCallback(() => {
+      setFormState(createFormInitialState(apiClient.listId, undefined));
+    }, [apiClient.listId]);
+
     const handleFlyoutClose = useCallback(() => {
       if (isSubmittingData) {
         return;
       }
 
       // `undefined` will cause params to be dropped from url
       setUrlParams({ id: undefined, show: undefined }, true);
-    }, [isSubmittingData, setUrlParams]);
+
+      if (isEditFlow) {
+        clearItem();
+      }
```

Comment:
I don't understand why you need to clear the item. The removal of the URL params will force the flyout to be closed.

FYI: there was a bug in the `setUrlParams()` above that my current pending PR fixes. It should be clearing out `itemId` not `id`

Reply:
When going from edit > close > new, the data from edit would still be there in the new form. Does your fix make the item no longer persist when flyout closes? if so, I can remove these clears.

Reply:
If this is happening, should be clear the form always we close it?

Reply:
Ok. This might have been a bug that I fixed. I was not clearing the `itemId` from the URL. If it is still happening, then I would like to dig into it. the only way this could happen (in my mind) is if the flyout is open and you click another items's edit on the list.  But on this list grid, you can't actually do that because the "edit" action is hidden behind the flyout.

_(I think this happens in Endpoint list (or was happening). Thats because in those cases, the flyout is not actually `unmounted`, but rather it gets new props)_

Reply:
Ok. I think I see the problem. It's a bug. I'm actually rendering the flyout regardless if in edit/create mode or not, it is just that it renders `null` if it determines it should not be opened. I'll will fix that in my PR. 

sorry for the issues @joeypoon

Reply:
If you want to apply the fix to your branch, its just adding a conditional in `artifact_list_age` - here: 

https://github.com/elastic/kibana/pull/126400/commits/8c7b72e24a6c85172a820f56e4ef8872c6c443e0

Reply:
I wrapped the flyout with `isFlyoutOpened` like the fix but I'm still seeing the same behavior.

Reply:
Ok. I think I found the bug (again) 😞 
The `ArtifactListPage` does not seem to be clearing out the `selectedItemForEdit`. You should be able to address this in your PR (temporarily ) by setting that prop to `undefined` inside of the `handleArtifactFlyoutOnSuccess()`.

For the real fix, I'm going to add an `onClose` callback prop in the Flyout and will use that to also ensure we clear it out when the user cancels/closes out the flyout. (I can't belive how bad of a decision it was to try to drive the flyout open/closed solely on URL params, especially when it can be triggered from the list as well 🤦 )